server/chess.py

```diff
@@ -271,14 +271,19 @@ def validate_king_move(
             if (not rook) or rook.has_moved:
                 return False
             for empty_file in empty_files:
-                if not self.get_piece(rank, empty_file):
+                if self.get_piece(rank, empty_file):
                     return False
-            return not self.hypothetical_check(
-                king.side, (king, rank, file), (rook, rank, rook_end)
-            )
+                elif self.hypothetical_check(
+                        king.side, (king, rank, empty_file)):
+                    return False
+            return True
         if (absolute_file_delta > 1) or (absolute_rank_delta > 1):
             return False
-        return True
+        if self.hypothetical_check(
+                    king.side, (king, rank, file)):
+            return False
```

Comment:
Moving into check is checked for in the global move validator, so no need to do it here.